public/app/features/trails/MetricOverviewScene.tsx

```diff
@@ -0,0 +1,232 @@
+    });
+
+    this.addActivationHandler(this._onActivate.bind(this));
+  }
+
+  private getVariable(): QueryVariable {
+    const variable = sceneGraph.lookupVariable(VAR_GROUP_BY, this)!;
+    if (!(variable instanceof QueryVariable)) {
+      throw new Error('Group by variable not found');
+    }
+
+    return variable;
+  }
+
+  private _onActivate() {
+    const metricScene = getMetricSceneFor(this);
+    this.updateMetadata();
+    this.updateLanguageProvider();
+
+    metricScene.subscribeToState((newState, oldState) => {
+      if (newState.metric !== oldState.metric) {
+        this.updateMetadata();
+      }
+    });
+
+    const variable = this.getVariable();
+    variable.subscribeToState((newState, oldState) => {
+      if (
+        newState.options !== oldState.options ||
+        newState.value !== oldState.value ||
+        newState.loading !== oldState.loading
+      ) {
+        const labels = this.getLabelOptions(this.getVariable()).filter((l) => l.value !== ALL_VARIABLE_VALUE);
+        this.setState({ labels, loading: variable.state.loading });
+        this.updateLabelValuesCount();
+      }
+    });
+
+    this.setState({ loading: variable.state.loading });
+  }
+
+  private updateLanguageProvider() {
+    const dsUid = sceneGraph.getAncestor(this, DataTrail).state.$variables?.getByName(VAR_DATASOURCE)?.getValue();
+    if (typeof dsUid === 'string') {
+      getDatasourceSrv()
+        .get(dsUid)
+        .then((ds) => {
+          const languageProvider: PrometheusLanguageProvider = ds.languageProvider;
+          this.setState({ languageProvider });
+        });
+    }
+  }
+
+  private updateMetadata() {
+    const metricScene = getMetricSceneFor(this);
+    const metric = metricScene.state.metric;
+    const dsUid = sceneGraph.getAncestor(this, DataTrail).state.$variables?.getByName(VAR_DATASOURCE)?.getValue();
+    if (typeof dsUid === 'string') {
+      getDatasourceSrv()
+        .get(dsUid)
+        .then((ds) => {
+          const langProvider: PrometheusLanguageProvider = ds.languageProvider;
+          if (langProvider.metricsMetadata) {
+            this.setState({ metadata: langProvider.metricsMetadata[metric] });
+          } else {
+            langProvider.start().then(() => {
+              this.setState({ metadata: langProvider.metricsMetadata?.[metric] });
+            });
+          }
+        });
+    }
+  }
+
+  private updateLabelValuesCount() {
+    this.state.labels.forEach((l) => {
+      if (l.value) {
+        const match = sceneGraph.interpolate(this, `${VAR_METRIC_EXPR}${VAR_FILTERS_EXPR}`, undefined, 'text');
+        this.state.languageProvider?.fetchSeriesValuesWithMatch(l.value, match).then((res) => {
+          this.setState({
+            labels: this.state.labels.map((label) =>
+              label.value === l.value ? { ...label, count: res.length } : label
+            ),
+          });
+        });
+      }
+    });
+  }
+
+  private getLabelOptions(variable: QueryVariable) {
```

Comment:
this function looks identical from the one in BreakdownScene, wonder if we can extract it to a util function somehow